src/Microsoft.DotNet.Darc/src/DarcLib/AzureDevOpsClient.cs

```diff
@@ -759,5 +763,153 @@ public Task CommitFilesAsync(List<GitFile> filesToCommit, string repoUri, string
+                HttpMethod.Put,
+                accountName,
+                projectName,
+                $"_apis/release/definitions/",
+                _logger,
+                body,
+                versionOverride: "5.0-preview.3",
+                baseAddressSubpath: "vsrm.");
+
+            return content.ToObject<AzureDevOpsReleaseDefinition>();
+        }
+
+        /// <summary>
+        ///     Remove all artifact sources of the release definition informed.
+        /// </summary>
+        /// <param name="accountName">Azure DevOps account name</param>
+        /// <param name="projectName">Project name</param>
+        /// <param name="releaseDefinition">Release definition to be modified</param>
+        public async void RemoveAllArtifactSourcesAsync(string accountName, string projectName, AzureDevOpsReleaseDefinition releaseDefinition)
+        {
+            releaseDefinition.Artifacts = new AzureDevOpsArtifact[0];
+
+            var body = JsonConvert.SerializeObject(releaseDefinition);
+
+            JObject content = await this.ExecuteRemoteGitCommandAsync(
+                HttpMethod.Put,
+                accountName,
+                projectName,
+                $"_apis/release/definitions/",
+                _logger,
+                body,
+                versionOverride: "5.0-preview.3",
+                baseAddressSubpath: "vsrm.");
+        }
+
+        /// <summary>
+        ///     Trigger a new release using the release definition informed. No change is performed
+        ///     on the release definition - it is used as is.
+        /// </summary>
+        /// <param name="accountName">Azure DevOps account name</param>
+        /// <param name="projectName">Project name</param>
+        /// <param name="releaseDefinition">Release definition to be updated</param>
+        public async void StartNewReleaseAsync(string accountName, string projectName, AzureDevOpsReleaseDefinition releaseDefinition)
+        {
+            var body = $"{{ \"definitionId\": {releaseDefinition.Id} }}";
+
+            JObject content = await this.ExecuteRemoteGitCommandAsync(
+                HttpMethod.Post,
+                accountName,
+                projectName,
+                $"_apis/release/releases/",
+                _logger,
+                body,
+                versionOverride: "5.0-preview.3",
+                baseAddressSubpath: "vsrm.");
+        }
+
+        /// <summary>
+        ///     Fetches an specific AzDO build based on its ID.
+        /// </summary>
+        /// <param name="accountName">Azure DevOps account name</param>
+        /// <param name="projectName">Project name</param>
+        /// <param name="buildId">Id of the build to be retrieved</param>
+        /// <returns>AzureDevOpsBuild</returns>
+        public async Task<AzureDevOpsBuild> GetBuildAsync(string accountName, string projectName, long buildId)
+        {
+            JObject content = await this.ExecuteRemoteGitCommandAsync(
```

Comment:
Wait... does "ExecuteRemoteGitCommandAsync" just execute random requests? Nothing about this seems to be Git related... That method should probably be renamed. I'm not sure why it's as complicated as it is... sending a single request with http client is pretty straightforward. :-)

Reply:
I just reused what was already available. If you think it's worth I might create another issue to keep track of that

Reply:
You changed it's meaning. Before it WAS only for git. Now it's not. You should rename it as part of this change where you changed it's meaning.

Reply:
Oh, now I got what you mean. Good catch. Thanks.